bson/json_util.py

```diff
@@ -810,6 +818,12 @@ def default(obj: Any, json_options: JSONOptions = DEFAULT_JSON_OPTIONS) -> Any:
         if json_options.datetime_representation == DatetimeRepresentation.LEGACY:
             return {"$date": millis}
         return {"$date": {"$numberLong": str(millis)}}
+    if isinstance(obj, DatetimeMS):
+        if json_options.datetime_representation == DatetimeRepresentation.ISO8601:
+            raise NotImplementedError()
+        elif json_options.datetime_representation == DatetimeRepresentation.LEGACY:
+            return {"$date": int(obj)}
+        return {"$date": {"$numberLong": int(obj)}}
```

Comment:
Converting from `DatetimeMS` to an ISO8601 date would require calculating the date to get the entire range of valid ISO8601 dates, does this mean this be a separate PR like you initially mentioned @ShaneHarvey? 

Going from `DatetimeMS -> datetime.datetime ->` ISO8601 would essentially require clamping.

Reply:
According to the Extended JSON spec, datetimes before the epoch (1970) or after year 9999 must always be encoded as `{"$date": {"$numberLong": <64-bit signed integer giving millisecs relative to the epoch, as a string>}}`: https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#conversion-table